cvprac/cvp_api.py

```diff
@@ -0,0 +1,500 @@
+                 'toId': dev['systemMacAddress'],
+                 'toIdType': 'netelement',
+                 'fromId': '',
+                 'nodeName': '',
+                 'fromName': '',
+                 'toName': dev['fqdn'],
+                 'childTasks': [],
+                 'parentTask': ''}]
+        self.log.debug('apply_configlets_to_device: saveTopology data:\n%s' %
+                       data)
+        self.clnt.post('/provisioning/saveTopology.do', data)
+
+    def remove_configlets_from_device(self, app_name, dev, del_configlets):
+        ''' Remove the configlets from the device.
+
+            Args:
+                app_name (str): The application name to use in info field.
+                dev (dict): The switch device dict
+                del_configlets (list): List of configlet name and key pairs
+        '''
+        self.log.debug('remove_configlets_from_device: dev: %s names: %s' %
+                       (dev, del_configlets))
+
+        # Get all the configlets assigned to the device.
+        configlets = self.get_configlets_by_device_id(dev['systemMacAddress'])
+
+        # Get a list of the names and keys of the configlets.  Do not add
+        # configlets that are on the delete list.
+        keep_names = []
+        keep_keys = []
+        for configlet in configlets:
+            key = configlet['key']
+            if next((ent for ent in del_configlets if ent['key'] == key),
+                    None) is None:
+                keep_names.append(configlet['name'])
+                keep_keys.append(key)
+
+        # Remove the names and keys of the configlets to keep and build a
+        # list of the configlets to remove.
+        del_names = []
+        del_keys = []
+        for entry in del_configlets:
+            del_names.append(entry['name'])
+            del_keys.append(entry['key'])
+
+        info = '%s Configlet Remove: from Device %s' % (app_name, dev['fqdn'])
+        info_preview = '<b>Configlet Remove:</b> from Device' + dev['fqdn']
+        data = [{'id': 1,
+                 'info': info,
+                 'infoPreview': info_preview,
+                 'note': '',
+                 'action': 'associate',
+                 'nodeType': 'configlet',
+                 'nodeId': '',
+                 'configletList': keep_keys,
+                 'configletNamesList': keep_names,
+                 'ignoreConfigletNamesList': del_names,
+                 'ignoreConfigletList': del_keys,
+                 'toId': dev['systemMacAddress'],
+                 'toIdType': 'netelement',
+                 'fromId': '',
+                 'nodeName': '',
+                 'fromName': '',
+                 'toName': dev['fqdn'],
+                 'childTasks': [],
+                 'parentTask': ''}]
+        self.log.debug('remove_configlets_from_device: saveTopology data:\n%s'
+                       % data)
+        self.clnt.post('/provisioning/saveTopology.do', data)
+
```

Comment:
I am not seeing a problem here on 2016.1.2.1. In the applicable cvprac test case it adds and removes a trivial configlet that only changes one line. This works fine without setting the "node*" parameters. I checked if there is an issue if the configlet changes the management IP address and ran that test and it worked fine without setting the "node*" parameters. I verified that CVP automatically picked up the new management address for the node.

I wonder if the "node*" parameters need to be set when ZTPing a new node. If that is the case then I can add them as optional parameters to the method call.